metricbeat/mb/event.go

```diff
@@ -116,14 +117,21 @@ func AddMetricSetInfo(module, metricset string, event *Event) {
 	if event.Host != "" {
 		info["host"] = event.Host
 	}
-	if event.Took > 0 {
-		info["rtt"] = event.Took / time.Microsecond
-	}
+
 	if event.Namespace != "" {
 		info["namespace"] = event.Namespace
 	}
 	info = common.MapStr{
 		"metricset": info,
+		"event": common.MapStr{
+			"dataset": fmt.Sprintf("%s.%s", module, metricset),
+		},
+	}
+
+	if event.Took > 0 {
+		// rtt is deprecated and will be removed in 7.0. Replaced by event.duration.
+		info.Put("metricset.rtt", event.Took/time.Microsecond)
+		info.Put("event.duration", event.Took/time.Nanosecond)
```

Comment:
Nit: here you have an innocuous race condition. The time.Nanosecond could happen later enough that `trunc(event.duration) != metricset.rtt`. It would be cleaner to populate `event.duration`, then truncate it to populate `metricset.rtt`.

Reply:
Not sure i can follow. event.Took is a value and it's not calculated here.

Reply:
Ah, didn't look up what `event.Took` was, sorry. Never mind this :-)